### PR TITLE
support oss object versioning

### PR DESCRIPTION
**Pull request by huiguangjun**

### CODE REVIEW

File: alicloud/data_source_alicloud_oss_buckets.go [hunk 1]
```diff
@@ -212,6 +212,19 @@ func dataSourceAlicloudOssBuckets() *schema.Resource {
 						},
 
 						"tags": tagsSchemaComputed(),
+
+						"versioning": {
+							Type:     schema.TypeMap,
+							Computed: true,
```
Comment (xiaozhu36) [inline alicloud/data_source_alicloud_oss_buckets.go]: I think it type should be a list.

File: alicloud/resource_alicloud_oss_bucket.go [hunk 2]
```diff
@@ -233,6 +233,29 @@ func resourceAlicloudOssBucket() *schema.Resource {
...
 			"tags": tagsSchema(),
+
+			"force_destroy": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+
+			"versioning": {
+				Type:     schema.TypeMap,
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_oss_bucket.go]: I think it type should be a list.

File: alicloud/resource_alicloud_oss_bucket.go [hunk 3]
```diff
@@ -969,6 +1028,34 @@ func resourceAlicloudOssBucketDelete(d *schema.ResourceData, meta interface{}) e
...
+							}
+
+							for _, object := range lor.ObjectVersions {
+								objectsToDelete = append(objectsToDelete, oss.DeleteObject{
+									Key:       object.Key,
+									VersionId: object.VersionId,
+								})
+							}
+
+							bucket.DeleteObjectVersions(objectsToDelete)
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_oss_bucket.go]: Does the method `DeleteObjectVersions ` return an error?

File: website/docs/r/oss_bucket.html.markdown [hunk 4]
```diff
@@ -167,6 +180,8 @@ The following arguments are supported:
 * `storage_class` - (Optional, ForceNew) The [storage class](https://www.alibabacloud.com/help/doc-detail/51374.htm) to apply. Can be "Standard", "IA" and "Archive". Defaults to "Standard".
 * `server_side_encryption_rule` - (Optional, Available in 1.45.0+) A configuration of server-side encryption (documented below).
 * `tags` - (Optional, Available in 1.45.0+) A mapping of tags to assign to the bucket. The items are no more than 10 for a bucket.
+* `versioning` - (Optional, Available in 1.45.0+ and ap-south-1 region) A state of versioning (documented below).
```
Comment (xiaozhu36) [inline website/docs/r/oss_bucket.html.markdown]: There is no need to add a tag `ap-south-1 region`. If there are some limitations, please add notes to show them.

File: alicloud/data_source_alicloud_oss_buckets.go (hunk 1 above)
Comment (huiguangjun) [reply alicloud/data_source_alicloud_oss_buckets.go]: done

File: alicloud/resource_alicloud_oss_bucket.go (hunk 3 above)
Comment (huiguangjun) [reply alicloud/resource_alicloud_oss_bucket.go]: add err  and check it.

File: website/docs/r/oss_bucket.html.markdown (hunk 4 above)
Comment (huiguangjun) [reply website/docs/r/oss_bucket.html.markdown]: done

File: alicloud/resource_alicloud_oss_bucket.go (hunk 2 above)
Comment (huiguangjun) [reply alicloud/resource_alicloud_oss_bucket.go]: done

**Pull request merged**